haskell-src/Concordium/Types/Queries.hs

```diff
@@ -1062,3 +1063,51 @@ data WinningBaker = WinningBaker
 --  record field is turned into the label @round@ in the corresponding JSON representation
 --  of the @PendingUpdate@.
 $(deriveJSON defaultOptions{fieldLabelModifier = firstLower . dropWhile isLower} ''WinningBaker)
+
+-- | A failing result of a dry run execution.
+data DryRunError
+    = -- | The current block state is undefined.
+      DryRunErrorNoState
+    | -- | The requested block was not found, so its state could not be loaded.
+      DryRunErrorBlockNotFound
+    | -- | The specified account was not found.
+      DryRunErrorAccountNotFound
+    | -- | The specified instance was not found.
+      DryRunErrorInstanceNotFound
+    | -- | The amount to mint would overflow the total CCD supply.
+      DryRunErrorAmountOverLimit {dreMaximumMintAmount :: !Amount}
+    | -- | The balance of the sender account is not sufficient to pay for the operation.
+      DryRunErrorBalanceInsufficient {dreRequiredAmount :: !Amount, dreAvailableAmount :: !Amount}
+    | -- | The energy supplied for the transaction is not sufficient to check the transaction
+      -- header.
+      DryRunErrorEnergyInsufficient {dreEnergyRequired :: !Energy}
+
+-- | A successful result of a dry run execution.
+-- These do not cover all successful results.
```

Comment:
What do you mean by this comment?

Reply:
There are some other successful results that are not covered by this type. The reason for this is that those cases have a (potentially) fallible Protobuf encoding, whereas these ones will always succeed.